runtime-sdk/src/context.rs

```diff
@@ -77,6 +81,22 @@ pub trait Context {
         self.mode() == Mode::SimulateTx
     }
 
+    /// Whether expensive queries are allowed based on local configuration.
+    ///
+    /// This method will always return `true` if `is_check_only` returns `false` to avoid any bugs
+    /// that would cause non-determinism in non-check-tx contexts.
```

Comment:
Because the output of this function depends on node-local configuration, using it during transaction execution would lead to non-determinism as it would be different between nodes.